src/bundles/csg/functions.ts

```diff
@@ -0,0 +1,823 @@
+);
+
+/**
+ * Primitive Shape of a rounded cube.
+ *
+ * @category Primitive
+ */
+export const rounded_cube: Shape = shapeSetOrigin(
+  new Shape(primitives.roundedCuboid({ size: [1, 1, 1] }))
+);
+
+/**
+ * Primitive Shape of a rounded cylinder.
+ *
+ * @category Primitive
+ */
+export const rounded_cylinder: Shape = shapeSetOrigin(
+  new Shape(primitives.roundedCylinder({ height: 1, radius: 0.5 }))
+);
+
+/**
+ * Primitive Shape of a geodesic sphere.
+ *
+ * @category Primitive
+ */
+export const geodesic_sphere: Shape = shapeSetOrigin(
+  new Shape(primitives.geodesicSphere({ radius: 0.5 }))
+);
+
+// [Variables - Colours]
+
+/**
+ * A hex colour code for black (#000000).
+ *
+ * @category Colour
+ */
+export const black: string = '#000000';
+
+/**
+ * A hex colour code for dark blue (#0000AA).
+ *
+ * @category Colour
+ */
+export const navy: string = '#0000AA';
+
+/**
+ * A hex colour code for green (#00AA00).
+ *
+ * @category Colour
+ */
+export const green: string = '#00AA00';
+
+/**
+ * A hex colour code for dark cyan (#00AAAA).
+ *
+ * @category Colour
+ */
+export const teal: string = '#00AAAA';
+
+/**
+ * A hex colour code for dark red (#AA0000).
+ *
+ * @category Colour
+ */
+export const crimson: string = '#AA0000';
+
+/**
+ * A hex colour code for purple (#AA00AA).
+ *
+ * @category Colour
+ */
+export const purple: string = '#AA00AA';
+
+/**
+ * A hex colour code for orange (#FFAA00).
+ *
+ * @category Colour
+ */
+export const orange: string = '#FFAA00';
+
+/**
+ * A hex colour code for light grey (#AAAAAA).
+ *
+ * @category Colour
+ */
+export const silver: string = '#AAAAAA';
+
+/**
+ * A hex colour code for dark grey (#555555).
+ *
+ * @category Colour
+ */
+export const gray: string = '#555555';
+
+/**
+ * A hex colour code for blue (#5555FF).
+ *
+ * @category Colour
+ */
+export const blue: string = '#5555FF';
+
+/**
+ * A hex colour code for light green (#55FF55).
+ *
+ * @category Colour
+ */
+export const lime: string = '#55FF55';
+
+/**
+ * A hex colour code for cyan (#55FFFF).
+ *
+ * @category Colour
+ */
+export const cyan: string = '#55FFFF';
+
+/**
+ * A hex colour code for light red (#FF5555).
+ *
+ * @category Colour
+ */
+export const rose: string = '#FF5555';
+
+/**
+ * A hex colour code for pink (#FF55FF).
+ *
+ * @category Colour
+ */
+export const pink: string = '#FF55FF';
+
+/**
+ * A hex colour code for yellow (#FFFF55).
+ *
+ * @category Colour
+ */
+export const yellow: string = '#FFFF55';
+
+/**
+ * A hex colour code for white (#FFFFFF).
+ *
+ * @category Colour
+ */
+export const white: string = '#FFFFFF';
+
+// [Functions]
+
+/**
+ * Union of the two provided shapes to produce a new shape.
+ *
+ * @param {Shape} a - The first shape
+ * @param {Shape} b - The second shape
+ * @returns {Shape} The resulting unioned shape
+ */
+export function union(a: Shape, b: Shape): Shape {
+  let newSolid: Solid = _union(a.solid, b.solid);
```

Comment:
I suppose it's a matter of programming style with its appropriate pros and cons, to never use `var`, use `const` for "true" constants (eg `export const DEFAULT_BORDER_RADIUS = "3px"`), and use `let` for the rest. I've been using it for a fair while and it can be seen in my other projects too. The updated extensive eslint config affords this.

Side note, after my finals, I have a few more TODOs planned, some of which are not filed as issues in our fork. One of them would be combining some values, such as ones extracted from the current stylesheets, into a single constants file. That is why there are a few standalone `//TODO` lines around the place, as some of the visual changes to the tab were a more impromptu decision made before the presentation

Reply:
Yeah I guess if it's part of your proposed eslint changes then it'd be fine, cause otherwise the rest of the modules repo follows the prefer `const` ruleset